ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/math/filters/ContinuousTransferFunction.java

```diff
@@ -0,0 +1,162 @@
+   {
+      this.name = name;
+      this.k = k;
+      this.numerator = numerator;
+      this.denominator = denominator;
+   }
+
+   public ContinuousTransferFunction(ContinuousTransferFunction[] severalContinuousTransferFunctions)
+   {
+      this("", severalContinuousTransferFunctions);
+   }
+
+   public ContinuousTransferFunction(String name, ContinuousTransferFunction[] severalContinuousTransferFunctions)
+   {
+      this.name = name;
+
+      int numOfTransferFunctionsToCascade = severalContinuousTransferFunctions.length;
+      ContinuousTransferFunction tf_total;
+
+      // Initialize total transfer function with first transfer function.
+      tf_total = severalContinuousTransferFunctions[0];
+
+      for (int i = 1; i < numOfTransferFunctionsToCascade; i++)
+      {
+         tf_total = CascadeTwoTransferFunctions(tf_total, severalContinuousTransferFunctions[i]);
+      }
+
+      // Set private variables to total transfer function variables;
+      this.k = tf_total.getGain();
+      this.numerator = tf_total.getNumerator();
+      this.denominator = tf_total.getDenominator();
+   }
+
+   /**
+    * Method cascades two transfer functions, H1(s) and H2(s) and returns a new transfer function H3(s).
+    * <pre>
+    *                            N1(s)      N2(s)        N3(s)
+    * H3(s) = H1(s)*H2(s) = k1 * ----- k2 * ----- = k3 * -----
+    *                            D1(s)      D2(s)        D3(s)
+    * </pre>
+    *
+    * @param H1 - Transfer Function 1
+    * @param H2 - Transfer Function 2
+    * @return H3 - Combined Transfer Function of 1 and 2 (H3(s) = H1(s)*H2(s))
+    */
+   private ContinuousTransferFunction CascadeTwoTransferFunctions(ContinuousTransferFunction H1, ContinuousTransferFunction H2)
+   {
+      ContinuousTransferFunction H3 = new ContinuousTransferFunction();
+
+      H3.setGain(H1.getGain() * H2.getGain());
+      H3.setNumerator(cascadePolynomials(H1.getNumerator(), H2.getNumerator()));
+      H3.setDenominator(cascadePolynomials(H1.getDenominator(), H2.getDenominator()));
+
+      return H3;
+   }
+
+   /**
+    * Method combines two polynomials, poly1(x) and poly2(x), into poly3(s).
+    *
+    * @param poly1
+    * @param poly2
+    * @return poly3 = poly1*poly2
+    */
+   private static double[] cascadePolynomials(double[] poly1, double[] poly2)
+   {
+      // Initialize new polynomial to new size.
+      double[] poly3 = new double[poly1.length + poly2.length - 1];
+
+      // Cycle through poly1 and poly2 to polynomials.
+      for (int i = 0; i < poly1.length; i++)
+      {
+         for (int j = 0; j < poly2.length; j++)
+         {
+            poly3[i + j] += poly1[i] * poly2[j];
+         }
+      }
+
+      return poly3;
+   }
+
+   public void setNumerator(double[] numerator)
+   {
+      this.numerator = numerator;
+   }
+
+   public void setDenominator(double[] denominator)
+   {
+      this.denominator = denominator;

```

Comment:
I would make these copy value, rather than store the pointer, so that it doesn't get modified unintentionally.